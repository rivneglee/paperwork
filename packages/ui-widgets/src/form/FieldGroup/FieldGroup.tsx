import React, { FunctionComponent, ReactElement } from 'react';
import 'react-toggle/style.css';

interface Props {
  label?: ReactElement | string;
  isRequired?: boolean;
  labelAccessory?: ReactElement;
}

const FieldGroup: FunctionComponent<Props> = ({
  label = '',
  isRequired = false,
  children,
  labelAccessory,
}) => (
    <div className="pw-field-group">
      <div className="pw-field-group__label">
        <label>{label}</label>
        <span className="pw-field-group__accessory">
          {labelAccessory}
          {isRequired && <span className="pw-field-group__asterisk">*</span>}
        </span>
      </div>
      <div className="pw-field-group__input">
        {children}
      </div>
    </div>
);

export default FieldGroup;
