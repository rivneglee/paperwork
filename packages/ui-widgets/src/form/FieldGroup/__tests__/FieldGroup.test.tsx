import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import FieldGroup from '../FieldGroup';

describe('FieldGroup', () => {
  const Label = () => <>Label</>;
  const Input = () => <>Input</>;

  it('should render field group with label', () => {
    const wrapper: ShallowWrapper = shallow(
      <FieldGroup label={<Label />}>
        <Input />
      </FieldGroup>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render field group without label', () => {
    const wrapper: ShallowWrapper = shallow(
      <FieldGroup>
        <Input />
      </FieldGroup>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
