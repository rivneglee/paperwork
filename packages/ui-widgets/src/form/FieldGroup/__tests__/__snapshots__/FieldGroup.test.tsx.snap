// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldGroup should render field group with label 1`] = `
<div
  className="pw-field-group"
>
  <div
    className="pw-field-group__label"
  >
    <label>
      <Label />
    </label>
    <span
      className="pw-field-group__accessory"
    />
  </div>
  <div
    className="pw-field-group__input"
  >
    <Input />
  </div>
</div>
`;

exports[`FieldGroup should render field group without label 1`] = `
<div
  className="pw-field-group"
>
  <div
    className="pw-field-group__input pw-field-group__input--full"
  >
    <Input />
  </div>
</div>
`;
